owncloudApp/src/main/java/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java

```diff
@@ -745,11 +746,51 @@ private void updateInputFormAccordingToServerCapabilities() {
                             mCapabilities.getFilesSharingPublicExpireDateDays()));
         }
 
+        // Set password label when opening the dialog
+        if (mReadOnlyButton.isChecked() &&
+                mCapabilities.getFilesSharingPublicPasswordEnforcedReadOnly().isTrue() ||
+                mReadWriteButton.isChecked() &&
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedReadWrite().isTrue() ||
+                mUploadOnlyButton.isChecked() &&
+                        mCapabilities.getFilesSharingPublicPasswordEnforcedUploadOnly().isTrue()) {
+            setPasswordEnforced();
+        }
+
+        // Set password label depending on the checked permission option
+        mPermissionRadioGroup.setOnCheckedChangeListener(
+                (group, checkedId) -> {
```

Comment:
Do you like this code style ?
If I would do it, I would join L760 and L761

Reply:
Done